### PR TITLE
Problem: no command to fix corrupted data in versiondb

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #1683

Solution:
- add fixdata command
- ignore version 0 in normal operation

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Release Notes

- **Bug Fixes**
	- Resolved an issue with corrupted data in the version database during startup.

- **Improvements**
	- Enhanced version database functionality to accept a string as the store name.
	- Introduced a feature to skip entries with a version of zero during data retrieval.
	- Added a command to fix incorrect data in the version database.
	- Implemented a new Timestamp API for improved data iteration.
	- Added support for opening the version database in read-only mode.

- **Tests**
	- Added new tests to verify the correct behavior of skipping version zero entries in the versioned store.

### CODE REVIEW

File: versiondb/tsrocksdb/store.go [hunk 1]
```diff
@@ -216,6 +232,62 @@ func (s Store) Flush() error {
...
+			return fmt.Errorf("invalid key length: %X", key)
+		}
+
+		ts := key[len(key)-TimestampSize:]
+		key = key[:len(key)-TimestampSize]
+		realKey := cloneAppend(prefix, key)
+
+		readOpts := grocksdb.NewDefaultReadOptions()
+		readOpts.SetTimestamp(ts)
+		oldValue, err := s.db.GetCF(readOpts, s.cfHandle, realKey)
```
Comment (mmsqe) [inline versiondb/tsrocksdb/store.go:273]: can oldValue be nil without error?

Comment (yihuang) [reply versiondb/tsrocksdb/store.go:273]: no, but it can be an empty slice, in that case `Data` returns `nil` slice.

**Review by mmsqe — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### Bug Fixes
 
 * [#1679](https://github.com/crypto-org-chain/cronos/pull/1679) Include no trace detail on insufficient balance fix.
+* [#1685](https://github.com/crypto-org-chain/cronos/pull/1685) Fix versiondb corrupted data on startup.
```
Comment (yihuang) [inline CHANGELOG.md:8]:
```suggestion
* [#1685](https://github.com/crypto-org-chain/cronos/pull/1685) Add command to fix versiondb corrupted data.
```

File: versiondb/tsrocksdb/store.go [hunk 3]
```diff
@@ -216,6 +232,63 @@ func (s Store) Flush() error {
...
+	defer iter.Close()
+
+	batch := grocksdb.NewWriteBatch()
+	defer batch.Destroy()
+
+	prefix := storePrefix(storeName)
+	for ; iter.Valid(); iter.Next() {
+		key := iter.Key()
+		if len(key) < TimestampSize {
+			return fmt.Errorf("invalid key length: %X", key)
```
Comment (mmsqe) [inline versiondb/tsrocksdb/store.go:264]:
```suggestion
			return fmt.Errorf("invalid key length: %X, store: %s", key, storeName)
```

File: versiondb/client/fixdata.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		Args:  cobra.ExactArgs(1),
+		Short: "Fix wrong data in versiondb, see: https://github.com/crypto-org-chain/cronos/issues/1683",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dir := args[0]
+			versionDB, err := tsrocksdb.NewStore(dir)
+			if err != nil {
+				return err
+			}
+
+			// see: https://github.com/crypto-org-chain/cronos/issues/1683
```
Comment (yihuang) [inline versiondb/client/fixdata.go:20]:
```suggestion
```

File: versiondb/tsrocksdb/store.go [hunk 5]
```diff
@@ -125,7 +125,7 @@ func (s Store) GetAtVersion(storeKey string, key []byte, version *int64) ([]byte
 // HasAtVersion implements VersionStore interface
 func (s Store) HasAtVersion(storeKey string, key []byte, version *int64) (bool, error) {
 	slice, err := s.GetAtVersionSlice(storeKey, key, version)
-	if err != nil {
+	if err != nil || slice == nil {
```
Comment (yihuang) [inline versiondb/tsrocksdb/store.go:128]: it's a bad idea that I return the `nil` in the first place, let me see if I can return a empty one.

**Pull request merged**